### PR TITLE
WIP: Add a DashboardResource

### PR DESCRIPTION
**Pull request by ankur12-1610**
This PR adds a new resource namely `DashboardResource` which is created in order to add reference of individual dashboards to their respective dashboard groups.

Although there is one ambiguity, for dashboard groups, should the `DashboardGroupResource` (which will be the new resource created) consist of a list of dashboards?

Do we really need a new resource for dashboard groups?

I'm thinking of a way in which iterating over dashboards and then storing values in a map in the frontend can be done as well, not sure if it is viable.

### CODE REVIEW

**Comment by sultan-duisenbay** [comment]
Hi Ankur and thanks for your PR. We are in the process of discussing the overall API structure and endpoints. Will keep you posted

File: pb/api/v1/data.proto [hunk 1]
```diff
@@ -11,8 +11,8 @@ import "pb/summary/summary.proto";
 
 service TestGridData {
   // GET /dashboards
-  // Lists dashboard names
-  rpc ListDashboard(ListDashboardRequest) returns (ListDashboardResponse) {}
+  // Lists dashboard names and the dashboard group they belong to
+  rpc ListDashboard(ListDashboardRequest) returns (ListDashboardAndDashboardGroupResponse) {}
```
Comment (chases2) [inline pb/api/v1/data.proto:15]: nit: avoid renaming this if possible. It's best if "ListDashboard" returns a "ListDashboardResponse"

**Review by chases2 — APPROVED** [review]
Looks good overall. One naming nit.

/hold

**Comment by sultan-duisenbay** [comment]
Covered by in #1215

**Pull request closed without merging**